keanu-project/src/main/java/io/improbable/keanu/vertices/bool/probabilistic/BernoulliVertex.java

```diff
@@ -63,7 +67,39 @@ public double logProb(BooleanTensor value) {
 
     @Override
     public Map<Long, DoubleTensor> dLogProb(BooleanTensor value) {
-        throw new UnsupportedOperationException();
+
+        if (!(probTrue instanceof Differentiable)) {
+            throw new UnsupportedOperationException("Probability is non-differentiable");
```

Comment:
Likelihood should probably be probability to avoid confusion with things like the L in MLE. "The probability of the Bernoulli being true must be differentiable"